nifi-nar-bundles/nifi-splunk-bundle/nifi-splunk-processors/src/main/java/org/apache/nifi/processors/splunk/QuerySplunkIndexingStatus.java

```diff
@@ -0,0 +1,236 @@
+        @ReadsAttribute(attribute = "splunk.send.at", description = "The time of sending the put request for Splunk.")})
+@SeeAlso(PutSplunkHTTP.class)
+public class QuerySplunkIndexingStatus extends SplunkAPICall {
+    private static final String ENDPOINT = "/services/collector/ack";
+
+    static final Relationship RELATIONSHIP_ACKNOWLEDGED = new Relationship.Builder()
+            .name("success")
+            .description("A FlowFile is transferred into this relationship when the acknowledgement was successful.")
+            .build();
+
+    static final Relationship RELATIONSHIP_UNACKNOWLEDGED = new Relationship.Builder()
+            .name("unacknowledged")
+            .description("A FlowFile is transferred into this relationship when the acknowledgement was not successful.")
+            .build();
+
+    static final Relationship RELATIONSHIP_UNDETERMINED = new Relationship.Builder()
+            .name("undetermined")
+            .description(
+                    "A FlowFile is transferred into this relationship when the acknowledgement state is not determined. " +
+                    "Flow files transferred into this relationship might be penalized! " +
+                    "This happens when Splunk returns with HTTP 200 but with false response for the acknowledgement id in the flow file attribute.")
+            .build();
+
+    static final Relationship RELATIONSHIP_FAILURE = new Relationship.Builder()
+            .name("failure")
+            .description("A FlowFile is transferred into this relationship when the acknowledgement was not successful.")
+            .build();
+
+    private static final Set<Relationship> RELATIONSHIPS = Collections.unmodifiableSet(new HashSet<>(Arrays.asList(
+            RELATIONSHIP_ACKNOWLEDGED,
+            RELATIONSHIP_UNACKNOWLEDGED,
+            RELATIONSHIP_UNDETERMINED,
+            RELATIONSHIP_FAILURE
+    )));
+
+    static final PropertyDescriptor TTL = new PropertyDescriptor.Builder()
+            .name("ttl")
+            .displayName("Maximum Waiting Time")
+            .description(
+                    "The maximum time the service tries to acquire acknowledgement confirmation for an index, from the point of registration. " +
+                    "After the given amount of time, the service considers the index as not acknowledged and moves it into the output buffer as failed acknowledgement.")
+            .defaultValue("1 hour")
+            .required(true)
+            .addValidator(StandardValidators.TIME_PERIOD_VALIDATOR)
+            .build();
+
+    static final PropertyDescriptor MAX_QUERY_SIZE = new PropertyDescriptor.Builder()
+            .name("max-query-size")
+            .displayName("Maximum Query Size")
+            .description(
+                    "The maximum number of acknowledgement identifiers the service query status for in one batch. " +
```

Comment:
This sentence seems to lack a predicate. Maybe
```suggestion
                    "The maximum number of acknowledgement identifiers the service query status(?) contains in one batch. " +
```